Source/Core/ManagedArray.js

```diff
@@ -0,0 +1,149 @@
+/*global define*/
+define([
+        './defaultValue',
+        './defined',
+        './defineProperties',
+        './Check'
+    ], function(
+        defaultValue,
+        defined,
+        defineProperties,
+        Check) {
+    'use strict';
+
+    /**
+     * A wrapper around arrays so that the internal length of the array can be manually managed.
+     *
+     * @alias ManagedArray
+     * @constructor
+     * @private
+     *
+     * @param {Number} [length=0] The initial length of the array.
+     */
+    function ManagedArray(length) {
```

Comment:
It's not. It's just an array that manually tracks its length internally. I thought this would be cleaner than having a new tracking index every time we want to manually manage the size of the raw array. Do you think this is unnecessary?